content/tutorials/generic/bootloader-reloading/bootloader-reloading.md

```diff
@@ -0,0 +1,112 @@
+- Click the **"Install"** button.
+- Wait for the installation to finish.
+- Click the **Close** button.
+- Select **File > Examples > Adafruit DAP library > samd21 > flash_from_SD** from the Arduino IDE's menus.
+- Change this line: ```#define SD_CS 4``` according to the Arduino pin connected to the SD CS pin. 
+
+    If your board has a built-in SD slot (e.g., [MKR Zero 9](https://store.arduino.cc/products/arduino-mkr-zero-i2s-bus-sd-for-sound-music-digital-audio-data)), then you can change this line:
+
+    ```
+    (!SD.begin(SD_CS)) { 
+    ```
+
+    to:
+
+    ```
+    if (!SD.begin()) {
+    ```
+
+- Select the correct board from the **Tools > Board** from the Arduino IDE's menus.
+- Select the correct port from the **Tools > Port** from the Arduino IDE's menus.
+- Select **Sketch > Upload** from the Arduino IDE's menus.
+- Wait for the upload to finish successfully.
+- Unplug the programmer Arduino board from your computer.
+- Plug the SD card into the SD slot connected to your Arduino board.
+- Connect the programmer Arduino board to the target Arduino board as follows:
+
+    | Programmer  | Target      |
+    | ----------- | ----------- |
+    | VCC         | +3V3        |
+    | 10          | SWDIO       |
+    | 9           | SWCLK       |
+    | GND         | GND         |
+    | 11          | RESETN      |
+
+    SWD pads on MKR boards other than MKR 1000:
+
+    ![](assets/SWDpadsMKR1000.png)
+
+    MKR1000 SWD header pinout:
+
+    ![](assets/SWDMKR1000header.png)
+
+    Nano 33 IoT SWD pads:
+
+    ![](assets/SWDpadsNano33IoT.png)
+
+- Plug the USB cable of the programmer Arduino board into your computer.
+- Select **Tools > Serial Monitor** from the Arduino IDE's menus. You should now see the Serial Monitor output showing the target board detected, and the bootloader file flashed to it successfully.
+- Unplug the programmer Arduino board from your computer.
+- Disconnect the programmer Arduino board from the target Arduino board.
+
+## Alternatives
+These are some alternatives to the "Adafruit DAP" method described above.
+
+### Using a CMSIS-DAP debug probe as the programmer
+
+If you have a CMSIS-DAP compliant debug probe, you can just do this instead:
+
+- Connect the debug probe to your Arduino board.
+- Select **Tools > Programmer > Atmel EDBG** from the Arduino IDE's menus.
+- Select **Tools > Burn Bootloader** from the Arduino IDE's menus. - The "Burn Bootloader" process should now finish successfully.
+
+You can use [this little open-source debugger](https://www.tindie.com/products/ataradov/cmsis-dap-compliant-swd-debugger/).
+
+### Using a J-Link as the programmer
+
+Segger J-Link debug probes (e.g., J-Link EDU Mini) can be used with the Adalink software:
+https://github.com/adafruit/Adafruit_Adalink.
+
+This is a fairly complex procedure, so we recommend against this option unless you already own a J-Link and don't have the supplies on hand for one of the other options.
```

Comment:
```suggestion
```

This is another case where I think it was maybe reasonable to mention in the context of a forum thread, but not appropriate for a tutorial. It would be worth writing separate tutorials for each of these alternative methods (though I found the software setup for the "Adalink" procedure to be very challenging when I performed it some years ago), then linking to each of the specific tutorials from an entry page so the reader can select the best method according to what hardware they happen to have on hand (e.g., if they have a CMSIS-DAP debug probe, that will be the easiest option by far).